src/server/mod.rs

```diff
@@ -1015,6 +1007,18 @@ where
                     ReplyCode::FileError,
                     "Failed to create directory",
                 )),
+                Event::InternalMsg(AuthSuccess) => {
+                    let mut session = session.lock()?;
+                    session.state = WaitCmd;
+                    Ok(Reply::new(
+                        ReplyCode::UserLoggedIn,
+                        "User logged in, proceed",
+                    ))
+                }
+                Event::InternalMsg(AuthFailed) => Ok(Reply::new(
+                    ReplyCode::NotLoggedIn,
+                    "Wrong username or password",
```

Comment:
Perhaps just say Auth Failed. If an auth back-end that does say 2FA fails on the 2FA step then the username and password may be completely correct but the authentication still failed.

Reply:
indeed, this far it was just a copy of the original error.